File: src/contexts/authenticating.js
var sha1 = require('sha1');
var bcrypt = require('bcrypt');

function hash(password, salt) {
  var pepper = "aMRbgUVL0fTilT";
  var intermediate = password + pepper + salt;
  return sha1(intermediate);
}

function salt() {
  return bcrypt.genSaltSync(10);
}

function Authenticating(password, salt, hashedPassword) {
  return hash(password, salt) === hashedPassword;
}

Authenticating.hash = hash;
Authenticating.salt = salt;
module.exports = Authenticating;

File: app/contexts/authenticating.rb
class Authenticating
  class << self
    def execute!(password, salt, hashed_password)
      hash(password, salt) == hashed_password
    end

    private

    def hash(password, salt)
      intermediate = password + pepper + salt
      Digest::SHA1.hexdigest(intermediate)
    end

    def pepper
      "aMRbgUVL0fTilT"
    end
  end
end